From 7ae4d5b60f5b7812e86d3c99ddd9adcdd45c307b Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Wed, 25 Apr 2018 03:08:24 +0200
Subject: [PATCH 1/2] hal: msim_voice_extn: Use an enum for keeping track of
 the current SIM

* Treat SIM2 as the "special" case rather than the fallback, while at it

Change-Id: I724918d2af4e9af59fce7a99caf45db370327bb8
---
 hal/voice_extn/msim_voice_extn.c | 33 ++++++++++++++++++--------------
 1 file changed, 19 insertions(+), 14 deletions(-)

diff --git a/hal/voice_extn/msim_voice_extn.c b/hal/voice_extn/msim_voice_extn.c
index 70012212e..1549ca086 100644
--- a/hal/voice_extn/msim_voice_extn.c
+++ b/hal/voice_extn/msim_voice_extn.c
@@ -1,5 +1,6 @@
 /*
- * Copyright (C) 2015 The CyanogenMod Project
+ * Copyright (C) 2015-2016 The CyanogenMod Project
+ * Copyright (C) 2018 The LineageOS Project
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -48,13 +49,18 @@
 extern int start_call(struct audio_device *adev, audio_usecase_t usecase_id);
 extern int stop_call(struct audio_device *adev, audio_usecase_t usecase_id);
 
-static int msim_phone_type = 1;
+typedef enum {
+    MSIM_SIM1 = 1,
+    MSIM_SIM2 = 2,
+} msim_phone_t;
+
+static msim_phone_t msim_phone = MSIM_SIM1;
 
 int msim_voice_extn_start_call(struct audio_device *adev)
 {
     audio_usecase_t usecase_id;
 
-    usecase_id = msim_phone_type == 1 ? USECASE_VOICE_CALL : USECASE_VOICE2_CALL;
+    usecase_id = msim_phone == MSIM_SIM2 ? USECASE_VOICE2_CALL : USECASE_VOICE_CALL;
     return start_call(adev, usecase_id);
 }
 
@@ -62,7 +68,7 @@ int msim_voice_extn_stop_call(struct audio_device *adev)
 {
     audio_usecase_t usecase_id;
 
-    usecase_id = msim_phone_type == 1 ? USECASE_VOICE_CALL : USECASE_VOICE2_CALL;
+    usecase_id = msim_phone == MSIM_SIM2 ? USECASE_VOICE2_CALL : USECASE_VOICE_CALL;
     return stop_call(adev, usecase_id);
 }
 
@@ -72,7 +78,7 @@ int msim_voice_extn_get_session_from_use_case(struct audio_device *adev,
 {
     int idx;
 
-    idx = msim_phone_type == 1 ? VOICE_SESS_IDX : VOICE2_SESS_IDX;
+    idx = msim_phone == MSIM_SIM2 ? VOICE2_SESS_IDX : VOICE_SESS_IDX;
     *session = &adev->voice.session[idx];
     return 0;
 }
@@ -93,17 +99,16 @@ int msim_voice_extn_set_parameters(struct audio_device *adev __unused,
                 property_get_int32(RADIO_PREFER_NETWORK_SLOT1)) {
             voice_slot = 0;
         }
-        if (strcmp(value, AUDIO_PARAMETER_VALUE_CP2)) {
-            msim_phone_type = voice_slot == 0 ? 1 : 0;
+        if (!strcmp(value, AUDIO_PARAMETER_VALUE_CP2)) {
+            msim_phone = voice_slot == 0 ? MSIM_SIM2 : MSIM_SIM1;
         } else {
-            msim_phone_type = voice_slot == 0 ? 0 : 1;
+            msim_phone = voice_slot == 0 ? MSIM_SIM1 : MSIM_SIM2;
         }
 #elif SAMSUNG_DUAL_SIM
-        msim_phone_type = property_get_int32(
-                strcmp(value, AUDIO_PARAMETER_VALUE_CP2) ?
-                AUDIO_PROPERTY_SEC_VSID1 : AUDIO_PROPERTY_SEC_VSID2) + 1;
+        msim_phone = property_get_int32(!strcmp(value, AUDIO_PARAMETER_VALUE_CP2) ?
+                AUDIO_PROPERTY_SEC_VSID2 : AUDIO_PROPERTY_SEC_VSID1) != 0 ? MSIM_SIM2 : MSIM_SIM1;
 #endif
-        ALOGV("%s: phone_type: %d", __func__, msim_phone_type);
+        ALOGV("%s: msim_phone=%d", __func__, msim_phone);
     }
 
     return 0;
@@ -114,7 +119,7 @@ int msim_voice_extn_is_call_state_active(struct audio_device *adev,
 {
     int idx;
 
-    idx = msim_phone_type == 1 ? VOICE_SESS_IDX : VOICE2_SESS_IDX;
+    idx = msim_phone == MSIM_SIM2 ? VOICE2_SESS_IDX : VOICE_SESS_IDX;
     *is_call_active = (adev->voice.session[idx].state.current == CALL_ACTIVE) ? true : false;
     return 0;
 }
@@ -122,6 +127,6 @@ int msim_voice_extn_is_call_state_active(struct audio_device *adev,
 int msim_voice_extn_get_active_session_id(struct audio_device *adev __unused,
                                          uint32_t *session_id)
 {
-    *session_id = msim_phone_type == 1 ? VOICE_VSID : VOICE2_VSID;
+    *session_id = msim_phone == MSIM_SIM2 ? VOICE2_VSID : VOICE_VSID;
     return 0;
 }
-- 
2.17.0

